pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -556,6 +556,11 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 		}
 		pod, containerSpec = restoredPod, restoredContainer
 	}
+
+	if containerSpec == nil {
```

Comment:
My suggestion is
    containerSpec = kubecontainer.GetContainerSpec(pod, containerName)
    if containerSpec == nil {
           .......